neard/src/lib.rs

```diff
@@ -174,6 +177,12 @@ pub fn apply_store_migrations(path: &String, near_config: &NearConfig) {
         let store = create_store(&path);
         set_store_version(&store, 16);
     }
+    #[cfg(feature = "protocol_feature_rectify_inflation")]
+    if db_version <= 16 {
+        info!(target: "near", "Migrate DB from version 16 to 17");
+        // version 16 => 17: add `timestamp` to `BlockInfo`
+        migrate_16_to_17(&path);
+    }
```

Comment:
Well, according to the https://github.com/near/nearcore/discussions/3582 discussion, we should better follow the option (2), in that case, I suggest we name the nightly migrations like `migrate_16_to_rectify_inflation` (this indicates which version of DB was expected to begin with, and which feature is going to be applied), and don't increment the DB version at the end of the migration, instead, apply all the nightly migrations as if they all "next" migration_16_*, and tick the DB version to something like +100000 at the end of all the migrations if nightly feature is enabled, thus we only be able to migrate DB once after stable database dump.